File: frontend/src/api/evaluation.ts
import { get, post } from '@/utils/request'

export interface Evaluation {
  id: number
  userId: number
  courseId: number
  studentId: number
  rating: number
  comment: string
  createdAt: string
}

export interface EvaluationRequest {
  courseId: number
  studentId?: number
  rating: number
  comment?: string
}

export function createEvaluation(userId: number, data: EvaluationRequest): Promise<Evaluation> {
  return post(`/parent/evaluations?userId=${userId}`, data)
}

export function getEvaluationById(id: number): Promise<Evaluation> {
  return get(`/parent/evaluations/${id}`)
}

export function getEvaluationsByCourseId(courseId: number): Promise<Evaluation[]> {
  return get(`/parent/evaluations/course/${courseId}`)
}

export function getCourseAverageRating(courseId: number): Promise<number> {
  return get(`/parent/evaluations/course/${courseId}/rating`)
}

File: backend/src/main/java/com/angli/online/entity/Evaluation.java
package com.angli.online.entity;

import com.baomidou.mybatisplus.annotation.*;
import lombok.Data;

import java.time.LocalDateTime;

@Data
@TableName("evaluations")
public class Evaluation {

    @TableId(type = IdType.AUTO)
    private Long id;

    private Long userId;

    private Long courseId;

    private Long studentId;

    private Integer rating;

    private String comment;

    @TableField(fill = FieldFill.INSERT)
    private LocalDateTime createdAt;

}